src/main/dynamo/Table.ts

```diff
@@ -42,11 +44,56 @@ export class Table<PK extends string = string, SK extends string = string> {
     return new GSIBuilder(this, name)
   }
 
-  addToEncryptionBlacklist(field: string) {
-    this.encryptionBlacklist.add(field)
+  registerGSI(gsi: GSI<string, string, any>) {
+    this.gsis.push(gsi)
+    this.encryptionBlacklist.add(gsi.partitionKeyName)
+    this.encryptionBlacklist.add(gsi.sortKeyName)
   }
 
   getEncryptionBlacklist(): Set<string> {
     return this.encryptionBlacklist
   }
+
+  asCreateTableInput(
+    billingMode: DynamoDB.Types.BillingMode
+  ): DynamoDB.Types.CreateTableInput {
+    const attributeSet = new Set([
+      this.partitionKeyName,
+      this.sortKeyName,
+      "model",
+      ...this.gsis.flatMap((_) => [_.partitionKeyName, _.sortKeyName]),
+    ])
+
+    const attributeDefinitions: DynamoDB.Types.AttributeDefinitions = Array.from(
+      attributeSet
+    ).map((attr) => ({
+      AttributeName: attr,
+      AttributeType: "S",
+    }))
+
+    return {
+      TableName: this.tableName,
+
+      KeySchema: [
+        { AttributeName: this.partitionKeyName, KeyType: "HASH" },
+        { AttributeName: this.sortKeyName, KeyType: "RANGE" },
+      ],
+
+      AttributeDefinitions: attributeDefinitions,
+      GlobalSecondaryIndexes: this.gsis.map(
+        ({ name, partitionKeyName, sortKeyName }) => ({
+          IndexName: name,
+          KeySchema: [
+            { AttributeName: partitionKeyName, KeyType: "HASH" },
+            { AttributeName: sortKeyName, KeyType: "RANGE" },
+          ],
+          Projection: {
+            ProjectionType: "ALL",
```

Comment:
Similar to the above, Beyonce assumes all indexes project all attributes. That makes the typing work out easier (i.e. you don't have to express partial model types). But it'd be nice to support that in the future.